L83-core-cpp-tls-key-export-feature.md

```diff
@@ -0,0 +1,93 @@
+Title
+----
+* Author(s): Vignesh Babu (Vignesh2208)
+* Approver: ctiller
+* Status: In Review
+* Implemented in: core, cpp
+* Last updated: 07/29/2021
+* Discussion at: https://github.com/grpc/grpc/issues/24944
+
+## Abstract
+
+This proposal discusses additions to the core/cpp public headers to support exporing of TLS Keys on the client and server side to aid in decryption of packet captures using tools such as Wireshark.
+
+## Background
+
+Currently unlike the GRPC Go implementation, GRPC C++ does not allow export of TLS keys at run time. This is needed to decrypt and analyse packet captures for debugging purposes. The keys when exported in files in a specific format, can be read by debugging tools like wireshark and they can automatically decrypt packet captures.
+
+### Related Proposals:
+
+N/A
+
+## Proposal
+
+The proposal extends the existing `grpc::experimental::TlsCredentialsOptions` class with methods to set configuration for supporting Tls Key export at run-time. The proposed changes
+include:
+
+* A `grpc_tls_key_log_format` enum is added to `include/grpc/grpc_security_constants.h`
+  ```
+  /** Supported TLS Keylogging formats **/
+  typedef enum {
+    // Default key logging formal is NSS compatible.
+    TLS_KEY_LOG_FORMAT_NSS = 0,
+    // More formats can be added here for future extensions if desired.
+    TLS_KEY_LOG_FORMAT_NONE
+  } grpc_tls_key_log_format;
+  ```
+
+* A `grpc::experimental::TlsKeyLoggerConfig` struct is defined in `include/grpcpp/security/tls_credentials_options.h` to support configuration for Key export. It currently has two members:
```

Comment:
Please document both the C-core API and the C++ API for the TlsCreds options.

Reply:
Done. I have added new sections to document them.